scala/scala-impl/test/org/jetbrains/plugins/scala/codeInspection/syntacticSimplification/PostfixUnaryOperationInspectionTest.scala

```diff
@@ -0,0 +1,199 @@
+package org.jetbrains.plugins.scala.codeInspection.syntacticSimplification
+
+import com.intellij.codeInspection.LocalInspectionTool
+import org.jetbrains.plugins.scala.codeInspection.{ScalaInspectionBundle, ScalaQuickFixTestBase}
+
+class PostfixUnaryOperationInspectionTest extends ScalaQuickFixTestBase {
+  override protected val classOfInspection : Class[_ <: LocalInspectionTool] = classOf[PostfixUnaryOperationInspection]
+  override protected val description = ScalaInspectionBundle.message("displayname.postfix.unary.operation")
+
+  private val hint = ScalaInspectionBundle.message("unary.operation.can.use.prefix.notation")
+
+  def testPostfixUnaryFunctionReference(): Unit = {
+    val original =
+      s"""import scala.language.postfixOps
+         |
+         |object A {
+         |  def unary_~(n: Int) = ~n
+         |  def f(): Int = 5 ${START}unary_~$END
+         |}
+      """.stripMargin
+
+    val result =
+      s"""import scala.language.postfixOps
+         |
+         |object A {
+         |  def unary_~(n: Int) = ~n
+         |  def f(): Int = ~5
+         |}
+      """.stripMargin
+
+    checkTextHasError(original)
+    testQuickFix(original, result, hint)
+  }
+
+  def testFunctionReferenceWithQualifier(): Unit = {
+    val original =
+      s"""object A {
+         |  def unary_~(n: Int) = ~n
+         |  def f(): Seq[Int] = Seq(1, 2, 3).map(A.${START}unary_~$END)
+         |}
+      """.stripMargin
+
+    val result =
+      s"""object A {
+         |  def unary_~(n: Int) = ~n
+         |  def f(): Seq[Int] = Seq(1, 2, 3).map(~A)
+         |}
+      """.stripMargin
```

Comment:
An interesting find I had during testing was that it is allowed to have the unary operator in front of an object containing the function. I presume there's some nice use-cases for this, but I was quite surprised when I saw it work in this example.

Reply:
Ahh yes, this is unexpected, indeed. I guess it's because the rewrite rule do happen on syntax level before any type checking/resolving.

Reply:
I guess so. On top of that, this does also compile and run with `unary_~` referenced through `~A`